Need to specify at least OSX 10.15 with CMAKE_OSX_DEPLOYMENT_TARGET

To fix the compiling error in OSX:
```
g2o/g2o/stuff/filesys_tools.cpp:36:31: error: 'path' is unavailable: introduced in macOS 10.15
```

Need to add `CMAKE_OSX_DEPLOYMENT_TARGET 10.15` to be minimum 10.15 to use `std::filesystem::path`

--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -1,5 +1,7 @@
 cmake_minimum_required(VERSION 3.14)
+
 set(CMAKE_LEGACY_CYGWIN_WIN32 0)
+set(CMAKE_OSX_DEPLOYMENT_TARGET "10.15" CACHE STRING "Minimum OS X deployment target")
 
 project(g2o)
 

